src/charm.py

```diff
@@ -196,22 +207,45 @@ def __init__(self, *args):
             self.on.check_config_action, self._on_check_config  # pyright: ignore
         )
 
-    def set_ports(self):
-        """Open necessary (and close no longer needed) workload ports."""
-        planned_ports = {
-            OpenedPort("tcp", self._ports.api),
-            OpenedPort("tcp", self._ports.ha),
-        }
-        actual_ports = self.unit.opened_ports()
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_unit_status)
 
-        # Ports may change across an upgrade, so need to sync
-        ports_to_close = actual_ports.difference(planned_ports)
-        for p in ports_to_close:
-            self.unit.close_port(p.protocol, p.port)
+    def _on_collect_unit_status(self, event: CollectStatusEvent):
+        # Resource limit
+        if not self.resources_patch.is_ready():
+            event.add_status(
+                MaintenanceStatus("[resource patch] Waiting for resource limit patch to apply")
+            )
+        if self.resources_patch.last_error:
+            event.add_status(BlockedStatus(f"[resource patch] {self.last_error}"))
```

Comment:
Should this be:
```suggestion
            event.add_status(BlockedStatus(f"[resource patch] {self.resources_patch.last_error}"))
```

Reply:
Also, I can see why you used the prefix here, however, will the error messages from the resource patch object already include enough context to avoid the need for it, and give a nicer status message?